federation/warder.go

```diff
@@ -0,0 +1,242 @@
+package federation
+
+import (
+	"database/sql"
+	"time"
+
+	btmTypes "github.com/bytom/protocol/bc/types"
+	"github.com/jinzhu/gorm"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/vapor/crypto/ed25519/chainkd"
+	"github.com/vapor/errors"
+	"github.com/vapor/federation/common"
+	"github.com/vapor/federation/config"
+	"github.com/vapor/federation/database/orm"
+	"github.com/vapor/federation/service"
+	vaporTypes "github.com/vapor/protocol/bc/types"
+)
+
+type warder struct {
+	position       uint8
+	xpub           chainkd.XPub
+	colletInterval time.Duration
+	db             *gorm.DB
+	txCh           chan *orm.CrossTransaction
+	mainchainNode  *service.Node
+	sidechainNode  *service.Node
+	remotes        []*service.Warder
+}
+
+func NewWarder(cfg *config.Config, db *gorm.DB, txCh chan *orm.CrossTransaction) *warder {
+	local, remotes := parseWarders(cfg)
+	return &warder{
+		position:       local.Position,
+		xpub:           local.XPub,
+		colletInterval: time.Duration(cfg.CollectMinutes) * time.Minute,
+		db:             db,
+		txCh:           txCh,
+		mainchainNode:  service.NewNode(cfg.Mainchain.Upstream),
+		sidechainNode:  service.NewNode(cfg.Sidechain.Upstream),
+		remotes:        remotes,
+	}
+}
+
+func parseWarders(cfg *config.Config) (*service.Warder, []*service.Warder) {
+	var local *service.Warder
+	var remotes []*service.Warder
+	for _, warderCfg := range cfg.Warders {
```

Comment:
for _, warderCfg := range cfg.Warders {
    w:=service.NewWarder(&warderCfg)
    if warderCfg.IsLocal {
         xxxxxx
   } else {
        xxxxxx
   }
}